@types/WebSocketClient.ts

```diff
@@ -1,19 +1,37 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-interface WebSocketClientInterface {
-    wsUrl: string;
-    disconnect(): void;
-}
+type SSLPinningConfiguration = {
+    enabled: boolean;
+    allowSelfSigned?: boolean;
+};
+
+type WebSocketClientConfiguration = {
+    headers?: ClientHeaders;
+    timeoutInterval?: number;
+    enableCompression?: boolean;
+    sslPinningConfiguration?: SSLPinningConfiguration;
+};
 
-type WebSocketCallbacks = {
-    onConnect(): void;
-    onDisconnect(): void;
-    onEvent(event: string, data: Record<string, string>): void;
+type WebSocketMessage =
+    | string
+    | WebSocketReadyState
+    | Record<string, string | number>;
+type WebSocketEvent = {
+    url: string;
+    message: WebSocketMessage;
 };
 
-type iOSWebSocketClientConfiguration = Record<string, string>;
-type AndroidWebSocketClientConfiguration = Record<string, string>;
-type WebSocketClientConfiguration =
-    | iOSWebSocketClientConfiguration
-    | AndroidWebSocketClientConfiguration;
+type WebSocketCallback = (event: WebSocketEvent) => void;
+
+interface WebSocketClientInterface {
+    url: string;
+    readyState: WebSocketReadyState;
+    send(data: string): void;
+    connect(): void;
+    close(): void;
+    onOpen(callback: WebSocketCallback): void;
```

Comment:
Just for consistency; we should do either `connect` / `onConnect` or `open` / `onOpen`?

MDN docs uses open; and the constructor just connects - so perhaps we should stick with `open()` and `onOpen()`?